tests/sources/fixtures/redis/fixture.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+# ruff: noqa: T201
+import os
+import random
+
+import redis.asyncio as redis
+
+from tests.commons import WeightedFakeProvider
+
+DATA_SIZE = os.environ.get("DATA_SIZE", "small").lower()
+_NUM_DB = {"small": 2, "medium": 4, "large": 16}
+NUM_DB = _NUM_DB[DATA_SIZE]
+RECORDS_TO_DELETE = 10
+EACH_ROW_ITEMS = 500
+ENDPOINT = "redis://localhost:6379/"
+
+fake_provider = WeightedFakeProvider(weights=[0.65, 0.3, 0.05, 0])
+
+
+async def inject_lines(redis_client, start, lines):
+    text = fake_provider.get_text()
+    rows = {}
+    for row_id in range(lines):
+        key = f"user_{row_id + start}"
```

Comment:
I still don't get the use of `start`. for db `0`, the key will be `user_0` to `user_499`, and for db `1`, the key will be `user_1` to `user_500`. Assuming the key can be the same in different databases, I don't see the point of using `start` here.

Reply:
Right, Actual point here is to make the key unique, which can be done by this: `key = f"user_{row_id}_{start}"` here `start` will make the key unique every time because start cannot be same.

Reply:
> If we remove `start` from key then key can be same in db0 and db1 which create same `_id` of a document i.e. duplicacy of IDs.

Interesting. So it's ok to have the same key in `db0` and `db1`? If so, we should remove `start` here, the same key in `db0` and `db1` should generate different `_id` in ES document.

Reply:
No it's important to be unique _id and for that I have changed it to `key = f"user_{row_id}_{start}"`

Reply:
A simple test shows it can have the same key in different dbs:
```
127.0.0.1:6379> select 0
OK
127.0.0.1:6379> set hello "value in db 0"
OK
127.0.0.1:6379> get hello
"value in db 0"
127.0.0.1:6379> select 1
OK
127.0.0.1:6379[1]> get hello
(nil)
127.0.0.1:6379[1]> set hello "value in db 1"
OK
127.0.0.1:6379[1]> get hello
"value in db 1"
127.0.0.1:6379[1]> select 0
OK
127.0.0.1:6379> get hello
"value in db 0"
```

In this case, I'd advise you to remove `start` here:
```python
key = f"user_{row_id}"
```

As `user_0` in db0 and db1 should result in different document id in ES.